### PR TITLE
window is displayed, but there are lots of errors in the console

### PR DESCRIPTION
**Pull request by matkuki**
Tested on Windows and Linux and both display `NimEdit`, but there are a lot of errors in the right `console` window:
image
```
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(28, 8) Error: cannot open file: macros
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(43, 12) Error: undeclared identifier: 'bindSym'
candidates (edit distance, scope distance); see '--spellSuggest': 
 (3, 4): 'binDir' [var declared in C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system\nimscript.nim(441, 5)]
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(43, 19) Error: attempting to call routine: 'bindSym'
  found 'bindSym' [unknown declared in C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(43, 12)]
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(43, 19) Error: attempting to call routine: 'bindSym'
  found 'bindSym' [unknown declared in C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(43, 12)]
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(43, 19) Error: expression 'bindSym' cannot be called
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(43, 19) Error: expression '' has no type (or is ambiguous)
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(43, 7) Error: 'let' symbol requires an initialization
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(44, 12) Error: undeclared identifier: 'newCall'
candidates (edit distance, scope distance); see '--spellSuggest': 
 (3, 4): 'readAll' [proc declared in C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system\io.nim(581, 6)]
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(44, 20) Error: expression 'fn' has no type (or is ambiguous)
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(44, 19) Error: attempting to call routine: 'newCall'
  found 'newCall' [unknown declared in C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(44, 12)]
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(44, 19) Error: attempting to call routine: 'newCall'
  found 'newCall' [unknown declared in C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\bitops.nim(44, 12)]
... and many more lines of these errors
```
Could I get a hint please, of where to look to start fixing this?

### CODE REVIEW

File: nimedit.nims [hunk 1]
```diff
@@ -6,10 +6,8 @@ switch("path", "$lib/../")
 
 --threads:on
 
---define: booting
---define:useStdoutAsStdmsg
-when defined(windows):
-  --app:gui
```
Comment (Araq) [inline nimedit.nims:12]: These switches are better than your new ones. ;-)

**Comment by Araq** [comment]
> Could I get a hint please, of where to look to start fixing this?

I think choosenim based installations are not really supported. Search for what paths are passed to "lib" in the NimScript layer.

File: nimedit.nims (hunk 1 above)
Comment (matkuki) [reply nimedit.nims:12]: Agreed, reverted with 811d2cb9d8c2c1568a0b101fca321b45c2bab7c1

**Comment by matkuki** [comment]
Adding the `lib/core` and `lib/system` to the `nimscript` search path fixes almost all except these 4 errors:
```
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\core\macros.nim(1168, 49) template/generic instantiation of `$` from here
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system\dollars.nim(145, 21) template/generic instantiation of `collectionToString` from here
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system\dollars.nim(129, 67) Error: type mismatch
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\core\macros.nim(1168, 49) template/generic instantiation of `$` from here
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system\dollars.nim(145, 21) template/generic instantiation of `collectionToString` from here
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system\dollars.nim(136, 13) template/generic instantiation of `addQuoted` from here
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system.nim(2930, 18) Error: type mismatch
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\strutils.nim(1497, 39) template/generic instantiation of `$` from here
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system\dollars.nim(107, 69) Error: type mismatch
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\strutils.nim(1497, 39) template/generic instantiation of `$` from here
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system\dollars.nim(111, 15) template/generic instantiation of `addQuoted` from here
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system.nim(2930, 18) Error: type mismatch
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\pure\strutils.nim(1497, 39) template/generic instantiation of `$` from here
C:\Users\matic\.choosenim\toolchains\nim-1.6.8\lib\system\dollars.nim(107, 69) Error: type mismatch
```
Any hints on why there are these type mismatches?

File: nimedit.nim [hunk 2]
```diff
@@ -411,8 +411,8 @@ proc loadOpenTabs(ed: Editor) =
           let suggested = parseInt(x[2])
           ed.con.hist[key] = CmdHistory(cmds: @[], suggested: suggested)
         of "histval":
-          doAssert(not key.len == 0)
-          ed.con.hist[key].cmds.add x[1]
+          if not key.len == 0:
```
Comment (Araq) [inline nimedit.nim]:
```suggestion
          if key.len != 0:
```

**Pull request closed without merging**